### PR TITLE
fix: fix test error for #246

### PR DESCRIPTION
**Pull request by fxxjdedd**
Now dateObject will always be a Date instance, so the judgment here needs to be changed.

for: https://github.com/yiminghe/async-validator/pull/246

### CODE REVIEW

**Comment by fxxjdedd** [comment]
@yiminghe 
I'm very sorry for missing the test case. I see that you have fixed this problem, but my logic will be a bit more concise.

File: src/rule/type.js [hunk 1]
```diff
@@ -34,12 +34,7 @@ const types = {
     }
   },
   date(value) {
-    return (
-      typeof value.getTime === 'function' &&
-      typeof value.getMonth === 'function' &&
-      typeof value.getYear === 'function' &&
-      !isNaN(value.getTime())
-    );
+    return value.toString() !== 'Invalid Date';
```
Comment (yiminghe) [inline src/rule/type.js:37]: is this  ecma standard?

Comment (fxxjdedd) [reply src/rule/type.js:37]: > toString()
This function returns a String value. The contents of the String are implementation-dependent, but are intended to represent the Date in the current time zone in a convenient, human-readable form.

No, it's not safe, my mistake. How about using only this line:
```
return !isNaN(value.getTime())
```

Comment (yiminghe) [reply src/rule/type.js:37]: current implementation is ok.

**Pull request closed without merging**